src/build/index.js

```diff
@@ -24,42 +24,43 @@ module.exports = async (argv) => {
   // Clean dist
   await del(path.join(process.cwd(), 'dist'))
 
-  // Run webpack
-  const webpack = await execa('webpack-cli', [
-    ...webpackConfig,
-    ...progress,
-    ...input,
-    ...forwardOptions
-  ], {
-    env: {
-      NODE_ENV: process.env.NODE_ENV || 'production',
-      AEGIR_BUILD_ANALYZE: argv.bundlesize,
-      AEGIR_NODE: argv.node,
-      AEGIR_TS: argv.ts
-    },
-    localDir: path.join(__dirname, '../..'),
-    preferLocal: true,
-    stdio: 'inherit'
-  })
+  if (argv.bundle) {
+    // Run webpack
+    await execa('webpack-cli', [
+      ...webpackConfig,
+      ...progress,
+      ...input,
+      ...forwardOptions
+    ], {
+      env: {
+        NODE_ENV: process.env.NODE_ENV || 'production',
+        AEGIR_BUILD_ANALYZE: argv.bundlesize,
+        AEGIR_NODE: argv.node,
+        AEGIR_TS: argv.ts
+      },
+      localDir: path.join(__dirname, '../..'),
+      preferLocal: true,
+      stdio: 'inherit'
+    })
 
-  if (argv.bundlesize) {
-    const stats = JSON.parse(fs.readFileSync(path.join(process.cwd(), 'dist/stats.json')))
-    const gzip = await gzipSize(path.join(stats.outputPath, stats.assets[0].name))
-    const maxsize = bytes(config.bundlesize.maxSize)
-    const diff = gzip - maxsize
+    if (argv.bundlesize) {
+      const stats = JSON.parse(fs.readFileSync(path.join(process.cwd(), 'dist/stats.json')))
+      const gzip = await gzipSize(path.join(stats.outputPath, stats.assets[0].name))
+      const maxsize = bytes(config.bundlesize.maxSize)
+      const diff = gzip - maxsize
 
-    console.log('Use http://webpack.github.io/analyse/ to load "./dist/stats.json".')
-    console.log(`Check previous sizes in https://bundlephobia.com/result?p=${pkg.name}@${pkg.version}`)
+      console.log('Use http://webpack.github.io/analyse/ to load "./dist/stats.json".')
+      console.log(`Check previous sizes in https://bundlephobia.com/result?p=${pkg.name}@${pkg.version}`)
 
-    if (diff > 0) {
-      throw new Error(`${bytes(gzip)} (▲${bytes(diff)} / ${bytes(maxsize)})`)
-    } else {
-      console.log(`${bytes(gzip)} (▼${bytes(diff)} / ${bytes(maxsize)})`)
+      if (diff > 0) {
+        throw new Error(`${bytes(gzip)} (▲${bytes(diff)} / ${bytes(maxsize)})`)
+      } else {
+        console.log(`${bytes(gzip)} (▼${bytes(diff)} / ${bytes(maxsize)})`)
+      }
     }
   }
 
-  if (hasTsconfig) {
+  if (argv.types && hasTsconfig) {
```

Comment:
I think it should probably fail if `--types` is passed while `tsconfig.json` isn't there instead of just silently ignoring the option.